### PR TITLE
Fix to check whether the name of role is built-in role or not

### PR DESCRIPTION
**Pull request by knanao**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/model/project.go [hunk 1]
```diff
@@ -413,7 +413,7 @@ func (p *Project) AddUserGroup(sso, role string) error {
 	if p.HasUserGroup(sso) {
 		return fmt.Errorf("%s is already being used. The SSO group must be unique", sso)
 	}
-	if !p.HasRBACRole(role) {
+	if !p.HasRBACRole(role) && !isBuiltinRBACRole(role) {
```
Comment (khanhtc1202) [inline pkg/model/project.go:416]: this looks like a leak to me, does it mean that there are some cases that `role` is not yet added into project RBAC roles but is being used as a role to add usergroup?

Comment (knanao) [reply pkg/model/project.go:416]: The build-in role is not added to a project hence we have to check whether the role which is used by the user group is a built-in role or not when adding a user group.

Comment (khanhtc1202) [reply pkg/model/project.go:416]: Ah get your point 👍 Then it's lgtm, thanks for the explanation.

**Review by khanhtc1202 — APPROVED** [review]
Nice catch

**Review by nghialv — APPROVED** [review]
👍

**Pull request merged**